cellxgene_schema_cli/scripts/gene_processing.py

```diff
@@ -14,258 +13,324 @@
-
-                # Add gene version if available from gene id
-                if feature in ["gene_id"]:
-                    if "." in target_features[i]:
-                        (feature_id, feature_version) = target_features[i].split(".")
-                    else:
-                        feature_id = target_features[i]
-                        feature_version = ""
-
-                    target_features[i] = feature_id
-                    current_features[feature.replace("id", "version")] = feature_version
-
-            output_to_print += ",".join(target_features + [str(current_length)]) + "\n"
-
-    write_gzip(output_to_print, output_file)
-
-
-def _get_gene_lengths_from_gtf(gtf_path: str) -> Dict[str, int]:
-    """
-    Parses a GTF file and calculates gene lengths, which are calculated as follows for each gene:
-       1. Get all different isoforms
-       2. Merge exons to create a set of non-overlapping "meta" exons
-       3. Sum the lengths of these "meta" exons
-
-    Code inspired from http://www.genemine.org/gtftools.php
-
-    :param str gtf_path: path to gzipped gtf file
-
-    :rtype  Dict[str]
-    :return A dictionary with keys being gene ids and values the corresponding length in base pairs
-    """
-
-    with gzip.open(gtf_path, "rb") as gtf:
-        # Dictionary of list of tuples that will store exon in bed format-like. Elements of the tuples will be the
-        # equivalent  fields from the bed format: chromosome, start, end, strand). Each list of tuples will correspond
-        # to one gene.
-
-        exons_in_bed = {}
-
-        for line in gtf:
-            line = line.decode("utf-8")
-
-            if line[0] == "#":
-                continue
-
-            line = line.rstrip().split("\t")
-
-            if line[2] != "exon":
-                continue
-
-            # Convert line to bed-like format: (chromosome, start, end, strand)
-            exon_bed = (line[0], int(line[3]) - 1, int(line[4]), line[6])
-
-            current_features = gtf_tools._get_features(line)
-            gene_id = current_features["gene_id"]
-            if gene_id in exons_in_bed:
-                exons_in_bed[gene_id].append(exon_bed)
+class GeneProcessingResult:
+    def __init__(self, gene_id: str, gene_name: str, gene_version: str, gene_length: str):
+        self.gene_id = gene_id  # ex: ENSG00000141510, should be unique
+        self.gene_name = gene_name  # ex: TP53, not necessarily unique
+        self.gene_version = gene_version
+        self.gene_length = gene_length
+
+
+class GeneProcessor:
+    def __init__(self):
+        # Global mapping of gene id to all the metadata we need to generate output files
+        self.gene_metadata: dict[str, GeneProcessingResult] = {}
+        # Mapping from description (ex: "mus musculus") to a list of gene ids
+        self.gene_ids_by_description: dict[str, list[str]] = {}
+        # Set of gene names that are duplicated
+        self.duplicated_gene_names = set()
```

Comment:
nit: delete--this class var isn't used, the uses are all method local

Reply:
good catch - removed :)